index.js

```diff
@@ -53,37 +68,44 @@ const parseArgs = (
   while (pos < argv.length) {
     let arg = argv[pos];
 
-    if (arg.startsWith('-')) {
+    if (StringPrototypeStartsWith(arg, '-')) {
       // Everything after a bare '--' is considered a positional argument
       // and is returned verbatim
       if (arg === '--') {
-        result.positionals.push(...argv.slice(++pos));
+        ArrayPrototypePush(
+          result.positionals,
+          ...ArrayPrototypeSlice(argv, ++pos)
+        );
```

Comment:
using spread allows hijacking with Array.prototype[Symbol.iterator]
```suggestion
        result.positionals = ArrayPrototypeConcat(
          result.positionals,
          ArrayPrototypeSlice(argv, ++pos)
        );
```